pkg/skaffold/cluster/minikube.go

```diff
@@ -0,0 +1,206 @@
+}
+
+type clientImpl struct{}
+
+func getClient() Client {
+	return clientImpl{}
+}
+
+func (clientImpl) IsMinikube(kubeContext string) bool {
+	// short circuit if context is 'minikube'
+	if kubeContext == constants.DefaultMinikubeContext {
+		return true
+	}
+	_, err := minikubeBinaryFunc()
+	if err != nil {
+		logrus.Debugf("Minikube cluster not detected: %v", err)
+		return false // minikube binary not found
+	}
+
+	if ok, err := matchNodeLabel(kubeContext); err != nil {
+		logrus.Debugf("failed to check minikube node labels: %v", err)
+	} else if ok {
+		logrus.Debugf("Minikube cluster detected: context %q nodes have minikube labels", kubeContext)
+		return true
+	}
+
+	if ok, err := matchProfileAndServerURL(kubeContext); err != nil {
+		logrus.Debugf("failed to match minikube profile: %v", err)
+	} else if ok {
+		logrus.Debugf("Minikube cluster detected: context %q has matching profile name or server url", kubeContext)
+		return true
+	}
+	logrus.Debugf("Minikube cluster not detected for context %q", kubeContext)
+	return false
+}
+
+func (clientImpl) MinikubeExec(arg ...string) (*exec.Cmd, error) {
+	return minikubeExec(arg...)
+}
+
+func minikubeExec(arg ...string) (*exec.Cmd, error) {
+	b, err := minikubeBinaryFunc()
+	if err != nil {
+		return nil, fmt.Errorf("getting minikube executable: %w", err)
+	}
+	return exec.Command(b, arg...), nil
+}
+
+func minikubeBinary() (string, error) {
+	execName := "minikube"
+	if found, _ := util.DetectWSL(); found {
+		execName = "minikube.exe"
+	}
+	filename, err := exec.LookPath(execName)
+	if err != nil {
+		return "", errors.New("unable to find minikube executable. Please add it to PATH environment variable")
+	}
+	if _, err := os.Stat(filename); os.IsNotExist(err) {
+		return "", fmt.Errorf("unable to find minikube executable. File not found %s", filename)
+	}
+	return filename, nil
+}
+
+func matchNodeLabel(kubeContext string) (bool, error) {
+	client, err := k8s.Client()
+	if err != nil {
+		return false, fmt.Errorf("getting Kubernetes client: %w", err)
+	}
+	opts := v1.ListOptions{
+		LabelSelector: fmt.Sprintf("minikube.k8s.io/name=%s", kubeContext),
+		Limit:         100,
+	}
+	l, err := client.CoreV1().Nodes().List(opts)
+	if err != nil {
+		return false, fmt.Errorf("listing nodes with matching label: %w", err)
+	}
+	return l != nil && len(l.Items) > 0, nil
+}
+
+// matchProfileAndServerURL checks if kubecontext matches any valid minikube profile
+// and for selected drivers if the k8s server url is same as any of the minikube nodes IPs
+func matchProfileAndServerURL(kubeContext string) (bool, error) {
+	config, err := getRestClientConfigFunc()
+	if err != nil {
+		return false, fmt.Errorf("getting kubernetes config: %w", err)
+	}
+	apiServerURL, _, err := rest.DefaultServerURL(config.Host, config.APIPath, schema.GroupVersion{}, false)
+
+	if err != nil {
+		return false, fmt.Errorf("getting kubernetes server url: %w", err)
+	}
+
+	logrus.Debugf("kubernetes server url: %s", apiServerURL)
```

Comment:
this definitely seems like `trace` level

Reply:
I changed most of the logs to `trace` level